acceptance-tests/src/test/java/gov/di_ipv_drivingpermit/pages/DrivingLicenceAPIPage.java

```diff
@@ -59,13 +74,123 @@ public void postRequestToSessionEndpoint() throws IOException, InterruptedExcept
+                        .setHeader("Content-Type", "application/json")
+                        .setHeader("session-id", SESSION_ID)
+                        .GET()
+                        .build();
+        String authCallResponse = sendHttpRequest(request).body();
+        LOGGER.info("authCallResponse = " + authCallResponse);
+        AuthorisationResponse deserialisedResponse =
+                objectMapper.readValue(authCallResponse, AuthorisationResponse.class);
+        AUTHCODE = deserialisedResponse.getAuthorizationCode().getValue();
+        LOGGER.info("authorizationCode = " + AUTHCODE);
+    }
+
+    public void postRequestToAccessTokenEndpointForDL(String criId)
+            throws IOException, InterruptedException {
+        String accessTokenRequestBody = getAccessTokenRequest(criId);
+        LOGGER.info("Access Token Request Body = " + accessTokenRequestBody);
+        String publicApiGatewayUrl = configurationService.getPublicAPIEndpoint();
+        LOGGER.info("getPublicAPIEndpoint() ==> " + publicApiGatewayUrl);
+        HttpRequest request =
+                HttpRequest.newBuilder()
+                        .uri(URI.create(publicApiGatewayUrl + "/token"))
+                        .setHeader("Accept", "application/json")
+                        .setHeader("Content-Type", "application/json")
+                        .POST(HttpRequest.BodyPublishers.ofString(accessTokenRequestBody))
+                        .build();
+        String accessTokenPostCallResponse = sendHttpRequest(request).body();
+        LOGGER.info("accessTokenPostCallResponse = " + accessTokenPostCallResponse);
+        Map<String, String> deserialisedResponse =
+                objectMapper.readValue(accessTokenPostCallResponse, new TypeReference<>() {});
+        ACCESS_TOKEN = deserialisedResponse.get("access_token");
+    }
+
+    public String requestDrivingLicenceCRIVC()
+            throws IOException, InterruptedException, ParseException {
+        String publicApiGatewayUrl = configurationService.getPublicAPIEndpoint();
+        HttpRequest request =
+                HttpRequest.newBuilder()
+                        .uri(URI.create(publicApiGatewayUrl + "/credential/issue"))
+                        .setHeader("Accept", "application/json")
+                        .setHeader("Content-Type", "application/json")
+                        .setHeader("Authorization", "Bearer " + ACCESS_TOKEN)
+                        .POST(HttpRequest.BodyPublishers.ofString(""))
+                        .build();
+        String requestDrivingLicenceVCResponse = sendHttpRequest(request).body();
+        LOGGER.info("requestDrivingLicenceVCResponse = " + requestDrivingLicenceVCResponse);
+        SignedJWT signedJWT = SignedJWT.parse(requestDrivingLicenceVCResponse);
+        return signedJWT.getJWTClaimsSet().toString();
+    }
+
+    public void validityScoreAndStrengthScoreInVC(Integer validityScore, Integer strengthScore)
```

Comment:
We have scoreIs in DrivingLicencePageObject could we reuse this method

Reply:
The scoreIs method in DrivingLicencePageObject looks for a webelement called 'JSONPayload' which we cannot use for the API tests.